src/main/webapp/app/service/firebase/firebase-gene-review-service.ts

```diff
@@ -63,162 +73,174 @@ export class FirebaseGeneReviewService {
       updateMetaData,
     );
 
-    const { hugoSymbol, pathFromGene } = parseFirebaseGenePath(firebasePath);
+    const { hugoSymbol } = parseFirebaseGenePath(firebasePath);
 
-    const updateObject = {
-      [pathFromGene]: updateValue,
-      [`${pathFromGene}_review`]: updatedReview,
-    };
+    let updateObject = this.getGeneUpdateObject(updateValue, updatedReview, firebasePath, uuid);
+    const metaUpdateObject = this.firebaseMetaService.getUpdateObject(!isChangeReverted, hugoSymbol, isGermline, uuid);
+    updateObject = { ...updateObject, ...metaUpdateObject };
 
-    // Make sure that there is a UUID attached
-    if (!uuid) {
-      uuid = generateUuid();
-      updateObject[`${pathFromGene}_uuid`] = uuid;
+    if (!shouldSave) {
+      return updateObject;
     }
 
     try {
-      await this.firebaseRepository.update(getFirebaseGenePath(isGermline, hugoSymbol), updateObject);
-      await this.firebaseMetaService.updateMeta(hugoSymbol, uuid, !isChangeReverted, isGermline);
+      await this.firebaseRepository.update('/', updateObject);
     } catch (error) {
-      throw new SentryError('Failed to update reviewable content', { firebasePath, currentValue, updateValue, review, uuid });
+      throw new SentryError('Failed to update reviewable content', updateObject);
     }
```

Comment:
Do you think we should return the `updateObject` even if we save? Feel like it would be nice to keep return value consistent